src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import type { User } from '../../types/User';
+import type { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then((userFromServer) => setUser(userFromServer))
+      .finally(() => setIsLoading(false));
```

Comment:
You can add catch block to handle errors.

Reply:
> You can add catch block to handle errors.

Good point 👍